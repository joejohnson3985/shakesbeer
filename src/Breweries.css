.brewery-header {
  display: flex;
  align-items: center;
}

.far {
  margin-right: 10px;
}
