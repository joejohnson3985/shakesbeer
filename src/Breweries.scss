@import './variables.scss';

.brewery-label {
  transition: box-shadow .3s ease-in-out;
  font-size: 15px;
  background-color: $secondary-color;
  margin: 20px 0;
  border-radius: 3px;
  border: 5px solid $secondary-color;

  &:hover {
    box-shadow: 0 0 14px rgba(33,33,33,.4);
  }
}

.brewery-header {
  display: flex;
  align-items: center;
  transition: background-color .5s ease-out;
  background-color: $primary-color;
  margin: 0;
  padding: 0px 20px;
  color: white;
  cursor: pointer;

  &:hover {
    background-color: $secondary-color;
  }

  h2 {
    flex: 15;
    padding: 20px;
    margin: 0px;
  }
}

.fa-star {
  font-size: 20px;
  color: #ffd68ba8;
  flex: 1;

  &:hover {
    color: white;
  }
}

.fas {
  color: white;
}
