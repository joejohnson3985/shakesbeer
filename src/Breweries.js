import React, { Component } from 'react';
import Beer from "./Beer.js"
import './Breweries.css';


class Breweries extends Component {
  constructor(props) {
    super(props);
    this.state = {
      dropDown: false,
      breweryName: '',
      beerList: [],
      favorite: false
    }
  }

    getTarget = (e) => {
      this.setState({breweryName: e.target.innerText}, () => {
        this.findBeers(); 
      }) 
      this.toggleDropDown()
    }

    findBeers = () => {
      let newBeers = this.props.dataset.find(brewery => {
        return this.state.breweryName === brewery.name;
      }).beers
      this.setState({beerList: newBeers});
    }

    toggleDropDown = () => {
      this.setState({dropDown: !this.state.dropDown})
    }

    toggleFavorite = (e) => {
      this.setState({favorite: !this.state.favorite});
      e.target.classList.toggle('fas');
    }
    
    render() {

      return (
        <div>
          <div className='brewery-header brewery-label'>
            <i className='far fa-star' onClick={this.toggleFavorite}></i>
            <h2 onClick={this.getTarget}>{this.props.name}</h2>
          </div>
          {
          this.state.dropDown ? (
              this.state.beerList.map(beer => {
                  return <Beer beers={this.state.beerList}
                               beerName={beer.name}
                               key={beer.FIELD1}
                          />
              })
          ) : (null)
          }
        </div>
      );
    }
}

export default Breweries;