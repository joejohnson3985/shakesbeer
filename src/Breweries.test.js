import React from 'react';
import Breweries from './Breweries.js';
import { shallow } from 'enzyme';


const mockData = [1,2,3,4];
const mockFunc = jest.fn();


describe('Breweries', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow (
        <Breweries name={mockData}
                   dataset={mockData}
        />
      )
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });


});