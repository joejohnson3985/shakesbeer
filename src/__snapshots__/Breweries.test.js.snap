// Jest Snapshot v1, https://goo.gl/fbAQLP


exports[`Breweries Should match the snapshot 1`] = `
<div
  className="brewery-label"
>
  <div
    className="brewery-header"
  >
    <i
      className="far fa-star"
      onClick={[Function]}
    />
    <h2
      onClick={[Function]}
    >
      The Lion Brewery
    </h2>
  </div>
  <div
    className="beer-dropdown hidden"
  />
</div>
`;
